static/app/utils/useCommitters.tsx

```diff
@@ -1,66 +1,31 @@
-import {useCallback, useEffect} from 'react';
-
-import {getCommitters} from 'sentry/actionCreators/committers';
-import {Client} from 'sentry/api';
-import CommitterStore, {getCommitterStoreKey} from 'sentry/stores/committerStore';
-import {useLegacyStore} from 'sentry/stores/useLegacyStore';
-import type {Committer, Organization, ReleaseCommitter} from 'sentry/types';
-import useApi from 'sentry/utils/useApi';
+import type {Committer} from 'sentry/types';
+import {QueryKey, useQuery} from 'sentry/utils/queryClient';
 
 import useOrganization from './useOrganization';
 
-interface Props {
+interface UseCommittersProps {
   eventId: string;
   projectSlug: string;
 }
 
-interface Result {
+interface CommittersResponse {
   committers: Committer[];
-  fetching: boolean;
-  // TODO(scttcper): Not optional on GA of release-committer-assignees flag
-  releaseCommitters?: ReleaseCommitter[];
 }
 
-async function fetchCommitters(
-  api: Client,
-  organization: Organization,
+const makeCommittersQueryKey = (
+  orgSlug: string,
   projectSlug: string,
   eventId: string
-) {
-  const repoData = CommitterStore.get(organization.slug, projectSlug, eventId);
-
-  if ((!repoData.committers && !repoData.committersLoading) || repoData.committersError) {
-    await getCommitters(api, {
-      orgSlug: organization.slug,
-      projectSlug,
-      eventId,
-    });
-  }
-}
-
-function useCommitters({eventId, projectSlug}: Props): Result {
-  const api = useApi();
-  const organization = useOrganization();
-  const store = useLegacyStore(CommitterStore);
-
-  const loadCommitters = useCallback(async () => {
-    await fetchCommitters(api, organization!, projectSlug, eventId);
-  }, [api, organization, projectSlug, eventId]);
-
-  useEffect(() => {
-    if (!organization) {
-      return;
+): QueryKey => [`/projects/${orgSlug}/${projectSlug}/events/${eventId}/committers/`];
+
+function useCommitters({eventId, projectSlug}: UseCommittersProps) {
+  const org = useOrganization();
+  return useQuery<CommittersResponse>(
+    makeCommittersQueryKey(org.slug, projectSlug, eventId),
+    {
+      staleTime: 600000,
```

Comment:
Should this data change over time? If not I'd be okay with `staleTime: Infinity` here